spec/controllers/idv/personal_key_controller_spec.rb

```diff
@@ -4,41 +4,42 @@
   include SamlAuthHelper
   include PersonalKeyValidator
 
-  def stub_idv_session
-    stub_sign_in(user)
-    idv_session.applicant = applicant
-    profile_maker = Idv::ProfileMaker.new(
-      applicant: applicant,
-      user: user,
-      user_password: password,
-    )
-    profile = profile_maker.save_profile(
-      fraud_pending_reason: nil,
-      gpo_verification_needed: false,
-      in_person_verification_needed: false,
-    )
-    idv_session.profile_id = profile.id
-    idv_session.personal_key = profile.personal_key
-    subject.idv_session.address_verification_mechanism = 'phone'
-    idv_session.vendor_phone_confirmation = true
-    idv_session.user_phone_confirmation = true
-    allow(subject).to receive(:idv_session).and_return(idv_session)
-  end
-
+  let(:applicant) { Idp::Constants::MOCK_IDV_APPLICANT_WITH_PHONE }
   let(:password) { 'sekrit phrase' }
   let(:user) { create(:user, :fully_registered, password: password) }
-  let(:applicant) { Idp::Constants::MOCK_IDV_APPLICANT_WITH_PHONE }
-  let(:profile) { subject.idv_session.profile }
-  let(:idv_session) do
-    Idv::Session.new(
-      user_session: subject.user_session,
-      current_user: user,
-      service_provider: nil,
-    )
-  end
+
+  # Most (but not all) of these tests assume that a profile has been minted
+  # from the data in idv_session. Set this to false to prevent this behavior
+  # and test the other way.
+  let(:mint_profile_from_idv_session) { true }
+
+  let(:address_verification_mechanism) { 'phone' }
+
+  let(:in_person_enrollment) { nil }
+
+  let(:idv_session) { subject.idv_session }
 
   before do
     stub_analytics
+    stub_attempts_tracker
+    stub_verify_steps_one_and_two(user, applicant: applicant)
```

Comment:
I am very suspicious of this method, and I think you don't need most of it. Once you use `stub_sign_in(user)`, you already have access to a user_session and an idv_session. (What does `verify_steps_one_and_two` even mean, anyway?) Doug and I have been chatting about ways to stub idv_session for different steps, and meanwhile we've been setting all the needed attributes.

Reply:
Yeah I don't like this method either